File: C_code/function_pointer/palindrome.c
#include<stdio.h>
int palindrome(int);
int main()
{
	int number,result;
	printf("enter a number:");
	scanf("%d",&number);
	int (*fp)(int)=palindrome;
	result=(*fp)(number);
	if(result != number)
		printf("given number is not a palindrome\n");
	else
		printf("given number is a palindrome\n");
}
int palindrome(int num)
{
	int rev=0,temp;
	while(num != 0)
	{
		temp = num % 10;
		rev=rev*10 + temp;
		num = num / 10;
	}
	return rev;

}

File: Python_Basic/palindrome.py
num=int(input("enter a number:"))
temp=num
rev=0
while(temp > 0):
    rev=rev*10+temp%10
    temp=temp//10

if(rev==num):
    print(num,"is a palindrome")
else:
    print(num,"is not a palindrome")
